### PR TITLE
RavenDB-13257 indexing performance  - duplicate items when replacemen…

### PR DESCRIPTION
**Pull request by ml054**
…t is applied

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -1106,6 +1106,20 @@ protected void ExecuteIndexing()
...
+                                            // this side-by-side index will be replaced in a second, notify about indexing success
+                                            // so we know that indexing batch is no longer in progress
+                                            DocumentDatabase.Changes.RaiseNotifications(new IndexChange
+                                            {
+                                                Name = Name,
+                                                Type = IndexChangeTypes.BatchCompleted
+                                            });
+                                            
+                                            var batchCompletedAction = DocumentDatabase.IndexStore.IndexBatchCompleted;
+                                            batchCompletedAction?.Invoke((Name, didWork));
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Indexes/Index.cs]: Let's extract this piece of code to a method. We have exactly the same a few lines code below.

Comment (ml054) [reply src/Raven.Server/Documents/Indexes/Index.cs]: done

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 2]
```diff
@@ -1238,6 +1228,22 @@ protected void ExecuteIndexing()
             }
         }
 
+        private void NotifyAboutCompletedBatch(bool didWork)
+        {
+            // this side-by-side index will be replaced in a second, notify about indexing success
+            // so we know that indexing batch is no longer in progress
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Indexes/Index.cs]: This comment should not be here but when `NotifyAboutCompletedBatch` it's called

Comment (ml054) [reply src/Raven.Server/Documents/Indexes/Index.cs]: done - sorry

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**